Makefile

```diff
@@ -30,4 +30,6 @@ test: build-test-container
 	            -e GRAPHQL_SERVER=https://app-interface.devshift.net/graphql \
 	            -e GRAPHQL_USERNAME=${USERNAME_PRODUCTION} \
 	            -e GRAPHQL_PASSWORD=${PASSWORD_PRODUCTION} \
+				-e QUAY_USER=${QUAY_USER} \
+				-e QUAY_TOKEN=${QUAY_TOKEN} \
```

Comment:
Could you try to pass QUAY_DOCKER_CONFIG_JSON and change https://github.com/app-sre/vault-manager/blob/master/tests/Dockerfile.tests#L24 to `CMD ["bash", "mkdir ~/.docker && echo $QUAY_DOCKER_CONFIG_JSON > ~/.docker/config.json && run-tests.sh"]` ?

That would be my guess.

Reply:
I gave this a shot, but I still haven't been able to get this working.

I'm building the container, mounting a volume with the docker config json into the test container, then running an interactive terminal inside the built image. 

Login is successful inside the container, but pulling any images in the private repository fails.

Reply:
pr_check.sh: add docker login to local config.json

```
DOCKER_CONF="$PWD/.docker"
mkdir -p "$DOCKER_CONF"
docker --config="${DOCKER_CONF}" login -u="${QUAY_USER}" -p="${QUAY_TOKEN}" quay.io
``

Makefile: mount created DOCKER_CONF to the test container

```
test: build-test-container
	@docker run -t \
	            --rm \
	            --net=host \
	            -v /var/run/docker.sock:/var/run/docker.sock \
				-v $(DOCKER_CONF):/docker \
	            -e GRAPHQL_SERVER=https://app-interface.devshift.net/graphql \
	            -e GRAPHQL_USERNAME=${USERNAME_PRODUCTION} \
	            -e GRAPHQL_PASSWORD=${PASSWORD_PRODUCTION} \
	            vault-manager-test
```

Dockerfile.tests: add ENV `DOCKER_CONFIG`

```
WORKDIR /tests
ENV DOCKER_CONFIG=/docker

CMD ["bash", "run-tests.sh"]
```

Run `sudo -E ./pr_check.sh`: fails due to docker credentials

```
sudo docker run -it -v /var/run/docker.sock:/var/run/docker.sock -v .../app-sre/vault-manager/.docker:/docker vault-manager-test:latest sh
```

Inside the test container, test logging into quay.io with CI credentials

```
sh-4.2# docker login -u <> -p <> quay.io/app-sre
Login Succeeded
```

Attempt to pull the Vault image

```
sh-4.2# docker run -d --net=host --name="vault-dev-server" --cap-add=IPC_LOCK -e 'VAULT_DEV_ROOT_TOKEN_ID=root' -p 8200:8200 -v /tmp/:/var/log/vault/:Z quay.io/app-sre/vault
Unable to find image 'quay.io/app-sre/vault:latest' locally
/usr/bin/docker-current: Error response from daemon: unauthorized: access to the requested resource is not authorized.
```

Reply:
Ok, I think I have resolution for the credentials issue. Adding `-v /var/run/docker.sock:/var/run/docker.sock` to `./tests/bats/helpers.bash` L5 in the docker run invocation does allow for pulling from private registries.